lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+module Asciidoctor
+# A built-in {Converter} implementation that generates HTML 5 output
+# that maximizes the use of semantic constructs.
+class Converter::SemanticHtml5Converter < Converter::Base
+  register_for 'semantic-html5', 'sem-html5'
+
+  def initialize backend, opts = {}
+    @backend = backend
+    if opts[:htmlsyntax] == 'xml'
+      syntax = 'xml'
+    else
+      syntax = 'html'
+    end
+    init_backend_traits basebackend: 'html', filetype: 'html', htmlsyntax: syntax, outfilesuffix: '.html', supports_templates: true
+  end
+
+  def convert_embedded node
+    node.content
+  end
+
+  def convert_paragraph node
+    attributes = html_attributes node.id, node.role, 'paragraph'
```

Comment:
I'm not convinced whether the "paragraph" class on the p is necessary. I fear it might leave the same bad taste as the div wrapper in the current HTML converter does. But I can understand the motivation for including it. It allows us to distinguish it from p tags used in other contexts. I wonder if simply marking with the role "block" would serve that purpose.

Note that Asciidoctor EPUB3 does not add a class to the p by default.

Reply:
Also note that the DocBook stylesheets do not add an implicit class to normal paragraphs either.

Reply:
Then maybe we shouldn't add a default role on paragraph blocks but rather always use specific roles when a `<p>` element is used in other contexts?

Reply:
> Then maybe we shouldn't add a default role on paragraph blocks but rather always use specific roles when a <p> element is used in other contexts?

Yes! That's a brilliant idea.